client/packages/common/src/plugins/hooks/usePluginElements.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { ComponentPluginType } from '../types';
+import { RecordWithId } from '../../types/utility';
+import { usePluginProvider } from '../components';
+import { PluginLoader } from '../components';
+
+export function usePluginElements<T extends RecordWithId>({
```

Comment:
this file was changes to a `.tsx` in order to use the `<PluginLoader />` 
the change is mainly to detect `isLoaded` and load async if required